src/FakeItEasy/Core/DefaultFakeManagerAccessor.cs

```diff
@@ -26,7 +26,21 @@ public FakeManager GetFakeManager(object proxy)
 
             if (result == null)
             {
-                throw new ArgumentException("The specified object is not recognized as a fake object.");
+                object tag = null;
+                ITaggable taggable = proxy as ITaggable;
+                if (taggable != null && taggable.Tag != null)
+                {
+                    tag = taggable.Tag;
+                }
+
+                if (tag == null)
+                {
+                    throw new ArgumentException(string.Format(ExceptionMessages.NotRecognisedAsFake, proxy.GetType().FullName));
+                }
+                else
+                {
+                    throw new ArgumentException(string.Format(ExceptionMessages.NotRecognisedAsFake, tag.GetType().FullName));

```

Comment:
I do not think we'll ever want to execute this branch. For two reasons.

1. the intent of #530 is to help users identify what kind of object they are attempting to use as a Fake when it isn't a Fake. The `tag` is not the object that the user supplied, so this could confuse thing
2. technically the `tag` could be something other than ` FakeManager`, but users would have to jump through some elaborate hoops to make it so. I think it would be impossible to do unintentionally, so we should never hit this case. I know we do in `FakeItEasy.Tests.Core.DefaultFakeManagerAccessorTests.Should_fail_when_getting_manager_from_object_where_a_wrong_typed_fake_manager_has_been_attached`, but that is, in my opinion an unrealistic test case. This isn't your fault. It's as if the test case were trying to trick you.

Either way, I think the correct action is to reference the type of `proxy` whenever `proxy` is not a Fake.
On the brighter side, this change will simplify the method considerably!